### ISSUE TITLE
Problems with webpack

### ISSUE THREAD

**Issue by cameronroe** [firebase/firepad#269]
I'm using webpack to load modules and using Firepad's Headless npm module. When I import the module it shows two errors:

```
Critical dependencies:
14:3292-3299 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/firepad/dist/firepad.min.js 14:3292-3299
```

```
Critical dependencies:
2463:61-68 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/firepad/dist/firepad.js 2463:61-68
```

I also can't use it within a React callback function. The function doesn't call the callback.

```
headless.setText(content, (err, committed) => {
    console.log(err, committed); // not called
})
```

**Comment by prescottprue** [comment]
@cameronroe I don't believe the warnings and this errors are related to headless not working.

## Things To Try
1. Make sure your rules allow you to do this write operation
2. Try passing a ref to Firepad instead of your url (i.e
```js
firebase.initializeApp(config);
const headless = new Firepad.Headless(firebase.database.ref.child('yourChild'));
```
3. Try calling this all in your `componentDidMount` (I have had this help with Firepad, especially if you are using something like CodeMirror or Ace since they both like window)

## Webpack Sidenote
I also needed the expose loaded to get Codemirror working with Firepad in Webpack like so:
```js
require('expose?CodeMirror!codemirror') // Needed for Firepad to load CodeMirror
```
This should also go in your `componentDidMount`

**Comment by zeratulinio** [comment]
Hello,

I have encountered the same error. It can be easily reproduced by creation new angular project with firepad. It looks like firepad can't be used anymore.

I am curious if someone managed to solve this somehow.

**Comment by prescottprue** [comment]
@zeratulinio Did you try the solution I posted? Still currently using this config and it is working for me.

**Comment by opensso** [comment]
I have it functionally working as well, but it would be nice to setup Angular application in the way it doesn't show any warniings.
@prescottprue @zeratulinio anyone has solution that would remove the warning?

**Comment by prescottprue** [comment]
@opensso You can try adding firepad to the `noParse` options of webpack (i.e. do not send it through the loaders at all), but that broke stuff I was trying to use it with.

There is also a way to silence webpack warnings, but I enjoy other warnings so I don't want them gone all together.
